CRM/Civicase/Upgrader/Steps/Step0015.php

```diff
@@ -12,13 +12,48 @@ class CRM_Civicase_Upgrader_Steps_Step0015 {
    *   True when the upgrader runs successfully.
    */
   public function apply() {
+    $this->addSingularLabelColumn();
+    $this->addSingularLabelToCaseCategories();
+
+    return TRUE;
+  }
+
+  /**
+   * Adds the singular label column to the case category instance table.
+   */
+  private function addSingularLabelColumn() {
     CRM_Core_DAO::executeQuery("
       ALTER TABLE civicrm_case_category_instance
       ADD COLUMN singular_label varchar(255) character set utf8mb4 COLLATE utf8mb4_unicode_ci
         DEFAULT NULL
     ");
+  }
 
-    return TRUE;
+  /**
+   * Adds the singular label value for each case category.
+   *
+   * If the case category ends in an S, it will remove it.
+   */
+  private function addSingularLabelToCaseCategories () {
+    $caseTypeCategories = civicrm_api3('OptionValue', 'get', [
+      'sequential' => '1',
+      'option_group_id' => 'case_type_categories',
+    ]);
+
+    foreach ($caseTypeCategories['values'] as $caseTypeCategory) {
+      $isLabelLastCharacterS = substr(strtolower($caseTypeCategory['label']), -1) === 's';
+      $singularLabel = $isLabelLastCharacterS
+        ? substr($caseTypeCategory['label'], -1)
```

Comment:
Will `substr($caseTypeCategory['label'], -1)` not evaluate to the last character? I thought we needed everything except the last character here?

Reply:
Horrible mistake. Thanks for spotting this.